mono/mini/aot-compiler.c

```diff
@@ -4344,6 +4344,58 @@ add_gc_wrappers (MonoAotCompile *acfg)
 	}
 }
 
+static gboolean enable_method_specialization;
+
+gboolean
+mono_aot_can_specialize (MonoMethod *method)
+{
+	if (!enable_method_specialization)
+		return FALSE;
+
+	if (!method)
+		return FALSE;
+
+	if (method->specialize_ran)
+		return method->can_specialize != 0;
+
+	if (method->wrapper_type != MONO_WRAPPER_NONE)
+		return FALSE;
+
+	// If it's not private, we can't specialize
+	if ((method->flags & METHOD_ATTRIBUTE_MEMBER_ACCESS_MASK) != METHOD_ATTRIBUTE_PRIVATE)
+		return FALSE;
+
+	// If it has the attribute disabling the specialization, we can't specialize
+	ERROR_DECL (cattr_error);
+	MonoCustomAttrInfo *cattr = mono_custom_attrs_from_method_checked (method, cattr_error);
+
+	gboolean success = TRUE;
+	if (!is_ok (cattr_error)) {
+		mono_error_cleanup (cattr_error);
+		success = FALSE;
+	} else if (!cattr) {
+		success = TRUE;
+	} else {
+		for (int i = 0; i < cattr->num_attrs; ++i) {
+			if (!cattr->attrs [i].ctor)
+				continue;
+
+			if (!strcmp (m_class_get_name (cattr->attrs [i].ctor->klass), "Reflected")) {
```

Comment:
Maybe check the namespace, too.  "Reflected" might be a pretty common name for attributes.

Reply:
Why not to use existing System.Runtime.CompilerServices.MethodImpl attribute with NoOptimization flag instead which this logic should honour anyway?

Reply:
Interesting. We don't seem to actually use that attribute at all right now outside of monodis.

https://gist.github.com/alexanderkyte/357ac0e85b37b078b17c511da154c473

I guess my concern was that by adding this attribute to a method, we would disable *other* optimizations. 

I can definitely re-use it if that's the case though.